pkg/flowaggregator/flowaggregator.go

```diff
@@ -180,7 +188,39 @@ func genObservationID() (uint32, error) {
 func (fa *flowAggregator) InitCollectingProcess() error {
 	var err error
 	var cpInput collector.CollectorInput
-	if fa.aggregatorTransportProtocol == AggregatorTransportProtocolTCP {
+	if fa.aggregatorTransportProtocol == AggregatorTransportProtocolTLS {
+		parentCert, privateKey, caCert, err := generateCACertKey()
+		if err != nil {
+			return fmt.Errorf("error when generating CA certificate: %v", err)
+		}
+		svc, err := fa.k8sClient.CoreV1().Services(flowAggregatorNamespace).Get(context.TODO(), flowAggregatorDeployment, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to get service %v: %v", flowAggregatorDeployment, err)
+		}
+		serverCert, serverKey, err := generateCertKey(parentCert, privateKey, true, svc.Spec.ClusterIP)
```

Comment:
I'm not really ok with this. The name should be used, not the ClusterIP. The Service may be re-created and be allocated with a different Service IP. Or the aggregator may not be deployed as a Service.

Reply:
With changes in go-ipfix v0.4.3, the input address of the collector will be resolved at go-ipfix side. Even when we provide the DNS name, at the time collector starts, it is using the resolved IP address so that the DNS name validation in certificate will not work. If we don't use Service IP, we can change go-ipfix code to not resolve DNS name before starting collector. How do you think about this solution?

Reply:
yes, please fix it in go-ipfix ASAP if you want this change to be included in Antrea v0.13

cc @srikartati